tests/test_quotient_manifold.py

```diff
@@ -0,0 +1,157 @@
+"""Unit tests for the quotient space."""
+
+import geomstats.backend as gs
+import geomstats.tests
+from geomstats.geometry.general_linear import GeneralLinear
+from geomstats.geometry.matrices import MatricesMetric
+from geomstats.geometry.quotient_manifold import FiberBundle, QuotientMetric
+from geomstats.geometry.spd_matrices import SPDMatrices, \
+    SPDMetricBuresWasserstein
+from geomstats.geometry.special_orthogonal import SpecialOrthogonal
+
+
+class TestQuotientSpace(geomstats.tests.TestCase):
+    def setUp(self):
+        gs.random.seed(1230)
+        n = 3
+        self.base = SPDMatrices(n)
+        self.base_metric = SPDMetricBuresWasserstein(n)
+        self.group = SpecialOrthogonal(n)
+        self.bundle = FiberBundle(
+            GeneralLinear(n), base=self.base, group=self.group)
+        self.quotient_metric = QuotientMetric(
+            self.bundle, ambient_metric=MatricesMetric(n, n))
+
+        def submersion(point):
+            return GeneralLinear.mul(point, GeneralLinear.transpose(point))
+
+        def tangent_submersion(tangent_vec, base_point):
+            product = GeneralLinear.mul(
+                base_point, GeneralLinear.transpose(tangent_vec))
+            return 2 * GeneralLinear.to_symmetric(product)
+
+        def horizontal_lift(tangent_vec, point, base_point=None):
+            if base_point is None:
+                base_point = submersion(point)
+            sylvester = gs.linalg.solve_sylvester(
+                base_point, base_point, tangent_vec)
+            return GeneralLinear.mul(sylvester, point)
+
+        self.bundle.submersion = submersion
```

Comment:
It is not very natural for a user to modify an object's method with this syntax.

Could we add:
- either a method `define_submersion` or `set_submersion` that takes a callable as input and basically performs the operation self.submersion = callable,
- or indicate in the docstring of `submersion` that this method can be replaced by the user?

Same for tangent_submersion, horizontal_lift and lift

Reply:
Yes I agree the user should be guided here:
- this might be a good solution if this class is to be instanciated as is, but I was thinking of it as an abstract class, and other child classes should be created which override these methods.
- in any case, either the docstring or a notebook will explain the different ways to use these classes: there is a minimum set of methods the user needs to explicitly define (e.g. align and vertical_projection as in Kendall)

Reply:
Got it, yes we can make it clear in the docstrings and notebooks.

Btw, we have not used the structure of Abstract Class (in Python) - using "raise NotImplementedError" instead - but we could have a look in the future:
https://www.tutorialspoint.com/abstract-base-classes-in-python-abc#:~:text=A%20class%20is%20called%20an,be%20implemented%20by%20its%20subclasses.